src/inmanta/server/agentmanager.py

```diff
@@ -635,20 +635,15 @@ def get_agent_client(self, tid: uuid.UUID, endpoint: str, live_agent_only: bool
         else:
             return None
 
-    async def are_agents_active(self, tid: uuid.UUID, endpoints: list[str]) -> bool:
+    async def expire_sessions_for_agents(self, env_id: uuid.UUID, endpoints: Set[str]) -> None:
```

Comment:
I think this is not entirely as intended: if the same endpoint exists both on the autostarted agent and another one, if will kill the session of both agents.

I would expect the kill to affect the agent's session only (i.e. all session you close here are intended to be the same, the agent process maintains a single session, not one per endpoint)

Am I correct that the major complication is that there is no clear way to distinguish which one is the autostarted one?

Reply:
An alternative solution here would be to implement the `on_connection_close` method of the request handler. 

The autostarted agent socket is always local, so it closes fast. But it is a whole different line of work going into the depths of the RPC framework. So I just mention it, but I would not necessarily do it.

Reply:
Discussed and concluded [on Slack](https://inmanta.slack.com/archives/CKRF0C8R3/p1715593454444139).